docs/src/route.ts

```diff
@@ -1,5 +1,10 @@
 const ROUTES = {
-  toolpadDocs: '/toolpad/getting-started/setup',
+  TOOLPAD_DOCS: '/toolpad/getting-started/setup',
+  // https://docs.netlify.com/site-deploys/overview/#deploy-contexts
+  TOOLPAD_SIGN_UP_URL:
+    process.env.CONTEXT !== 'production' || process.env.NODE_ENV !== 'production'
```

Comment:
Why not the same logic as https://github.com/mui/material-ui/blob/1c9b514738037e81fcbc1b434191cab128eb64ed/docs/src/config.ts#L3?

But then, we also have https://github.com/mui/material-ui/blob/1c9b514738037e81fcbc1b434191cab128eb64ed/docs/next.config.js#L197, why not even trash DEPLOY_CONTEXT to use STAGING?

Reply:
I feel that the `CONTEXT` variable is better suited to determining what the environment is ("production", "deploy-preview" or "branch-deploy") and it is documented clearly with this purpose as well: https://docs.netlify.com/configure-builds/environment-variables/#build-metadata, whereas the `PULL_REQUEST` variable is documented as being useful to check "whether the build is from a pull/merge request (true) or not (false).": https://docs.netlify.com/configure-builds/environment-variables/#git-metadata

Reply:
We frequently use `CONTEXT="branch-deploy"` for hosting content in production. e.g. https://v4.mui.com/. So I would consider this as production. My main concern is that we don't have 10 different ways to determine production, it's too painful to debug down the line. I think that we should standardize on one way.

Reply:
Which way do you propose to be adopted as the standard?

Reply:
> Which way do you propose to be adopted as the standard?

First, I would look at what we already use to determine production:

1. `process.env.NODE_ENV === 'production' && !process.env.PULL_REQUEST` [in config.js](https://github.com/mui/material-ui/blob/1c9b514738037e81fcbc1b434191cab128eb64ed/docs/src/config.ts#L3)
2. `process.env.NODE_ENV === 'production' && !process.env.PULL_REQUEST` [in document.js](https://github.com/mui/material-ui/blob/1c9b514738037e81fcbc1b434191cab128eb64ed/docs/pages/_document.js#L32)
3. `process.env.REPOSITORY_URL !== undefined && /mui-org\/material-ui-docs$/.test(process.env.REPOSITORY_URL);` [in next.js](https://github.com/mui/material-ui/blob/1c9b514738037e81fcbc1b434191cab128eb64ed/docs/next.config.js#L197)

I believe 3. is wrong, we should be using the environment variables to fill these links, and that's it:

<img width="303" alt="Screenshot 2022-08-25 at 11 32 20" src="https://user-images.githubusercontent.com/3165635/186629834-83451ee0-2fc7-4707-8c12-030708d6f2bb.png">

And that's it. So I think that we should pick 1. and 2.

Reply:
Now, I think that the challenge is how do we make this a standard? I think that we should add a new variable inside the `env: {` path of the `next.config.js`. Maybe having an `DEPLOY_ENV` would make sense. Something that would replicate the [CONTEXT](https://docs.netlify.com/configure-builds/environment-variables/#build-metadata) but tailored for our use case. I would imagine 3 values: `production`, `staging`, `development`.

Reply:
In general, my recommendation would be to not designate "environments", but instead use "flags" with sensible defaults, and supply the right flags when you deploy. i.e. don't do

```tsx
if (process.env.DEPLOY_TARGET === 'pull-request') {
  doFoo()
  doBar()
}
```
but do
```tsx
if (process.env.DO_FOO) {
  (()
}
if (process.env.DO_BAR) {
  doBar()
}
```

This essentially decouples the definition of how an environment behaves from the actual code. Some of the advantages are that it's much faster to reconfigure and introducing a new environment doesn't require a code change.

`NODE_ENV` is a widely accepted standard to designate between production and not production, and can be kept for minification and enabling debugging features.

Reply:
@Janpot Sure, we could add this abstraction. However, How would we define `DO_FOO`? I assume we need `DEPLOY_TARGET` beforehand.

Reply:
> However, how would we define DO_FOO? I assume this variable would be infered from DEPLOY_TARGET.

Yes, eventually the same thing is happening, somewhere there needs to be a translation between a target identifier, and the features it unlocks. The question is: where and when do we want this translation to take place. Is it in the code? Or external to the code? At build time? Or at deploy time?